src/plugins/intel_cpu/src/cpu_memory.h

```diff
@@ -364,8 +353,90 @@ class Memory : public IMemory {
     }
 };
 
+class StringMemory : public IMemory {
+public:
+    using OvString = ov::element_type_traits<ov::element::string>::value_type;
+
+    class StringMemoryMngr : public IMemoryMngr {
+    public:
+        StringMemoryMngr() : m_data(nullptr, release) {}
+        OvString* getStringPtr() const noexcept;
+        void setExtStringBuff(OvString* ptr, size_t size);
+        size_t getStrLen() const noexcept;
+
+        void* getRawPtr() const noexcept override;
+        void setExtBuff(void* ptr, size_t size) override;
+        bool resize(size_t size) override;
```

Comment:
This call has different semantics from the `IMemoryMngr` interface as the size is considered to be in elements of size `string`. I'm even thinking whether we really need to inherit `IMemoryMngr` interface, as almost all the interface calls has different semantics?

Reply:
Done